modules/images/webp-uploads/load.php

```diff
@@ -35,5 +35,186 @@ function webp_uploads_filter_image_editor_output_format( $output_format, $filena
 
 	return $output_format;
 }
-add_filter( 'image_editor_output_format', 'webp_uploads_filter_image_editor_output_format', 10, 3 );
 
+/**
+ * Hook called by `wp_generate_attachment_metadata` to create the `sources` property for every image
+ * size, the sources' property would create a new image size with all the mime types specified in
+ * `webp_uploads_valid_image_mime_types`. If the original image is one of the mimes from
+ * `webp_uploads_valid_image_mime_types` the image is just added to the `sources` property and  not
+ * created again. If the uploaded attachment is not a valid image this function does not alter the
+ * metadata of the attachment, on the other hand a `sources` property is added.
+ *
+ * @since n.e.x.t
+ *
+ * @see   wp_generate_attachment_metadata
+ * @see   webp_uploads_valid_image_mime_types
+ *
+ * @param array $metadata      An array with the metadata from this attachment.
+ * @param int   $attachment_id The ID of the attachment where the hook was dispatched.
+ *
+ * @return array An array with the updated structure for the metadata before is stored in the database.
+ */
+function webp_uploads_create_sources_property( array $metadata, $attachment_id ) {
+	// This should take place only on the JPEG image.
+	$valid_mime_types = webp_uploads_valid_image_mime_types();
+
+	// Not a supported mime type to create the sources property.
+	if ( ! array_key_exists( get_post_mime_type( $attachment_id ), $valid_mime_types ) ) {
+		return $metadata;
+	}
+
+	// All subsizes are created out of the original image.
+	$file = wp_get_original_image_path( $attachment_id, true );
+
+	// File does not exist.
+	if ( ! file_exists( $file ) ) {
+		return $metadata;
+	}
+
+	// Prevent to convert JPEG to WebP if we are creating JPEG versions of the image.
+	remove_filter( 'image_editor_output_format', 'webp_uploads_filter_image_editor_output_format' );
```

Comment:
I'm not sure about this here - shouldn't it be in a conditional? If the user uploads a JPEG image, we _do_ need to create WebP sub-sizes here by default, so why remove the filter?

Reply:
Right, the way the filter works it would convert any JPEG to WebP, so in the scenario when a user uploads a WebP image we need to generate JPEG subsizes instead. 

If the filter remains as it is, when we aim to create a JPEG effectively is going to create a WebP image instead, so when this is the case we need to make sure that does not happen.

Agree I think this can be more robust in terms of only removing it when we are actually creating a JPEG image.